zulip/zulip/__init__.py

```diff
@@ -1072,17 +1072,20 @@ def deregister(self, queue_id: str, timeout: Optional[float] = None) -> Dict[str
             timeout=timeout,
         )
 
-    def get_profile(self, request: Optional[Dict[str, Any]] = None) -> Dict[str, Any]:
+    def get_profile(self, request: Optional[Dict[str, Any]] = None, **kwargs: Any) -> Dict[str, Any]:
         '''
             Example usage:
 
             >>> client.get_profile()
             {u'user_id': 5, u'full_name': u'Iago', u'short_name': u'iago', ...}
         '''
+        if request is not None:
+            logging.warning('Passing parameters as a dictionary will be deprecated in future versions.')
+
         return self.call_endpoint(
             url='users/me',
             method='GET',
-            request=request,
+            request=request or kwargs,
```

Comment:
This ignores `kwargs` if `request` is passed, which may be confusing and doesn't follow the logic @andersk and I discussed in the issue.